### PR TITLE
SONARIAC-887 Rule S6379: ARM Enabling Azure resource-specific admin accounts is security-sensitive - Microsoft.ContainerRegistry/registries

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

File: iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "resources": [
+    {
+      "name": "Noncompliant: adminUserEnabled is set to true",
+      "type": "Microsoft.ContainerRegistry/registries",
+      "apiVersion": "2019-05-01",
+      "properties": {
+        "adminUserEnabled": true
+      }
+    },
+
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.json:13]: Nitpick, empty line

File: iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      }
+    },
+    {
+      "name": "Compliant: adminUserEnabled is not set",
+      "type": "Microsoft.ContainerRegistry/registries",
+      "apiVersion": "2019-05-01",
+      "properties": {
+        "__comment": "Compliant: default value for adminUserEnabled is false"
+      }
+    }
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.json:37]: Nitpick. You can add the case when `adminUserEnabled ` is not a string literal

File: iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.bicep [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    adminUserEnabled: false
+  }
+}
+
+resource acrAdminInvalidValue 'Microsoft.ContainerRegistry/registries@2021-09-01' = {
+  properties: {
+    adminUserEnabled: 'true'
+  }
+}
+
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.bicep:19]: Nitpick. You can add the case when `adminUserEnabled ` is not a string literal

**Review by mstachniuk — APPROVED** [review]
LGTM and please see my comments

File: iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.json (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.json:13]: I usually add this empty line to separate between NonCompliant and Compliant test cases :smile:

File: iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.json (hunk 2 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.json:37]: Mmhh it is expecting to find a `BooleanLiteral`, I added a case with an unexpected type `StringLiteral`. Do you mean you would like another case with an unexpected type which is not a `TextTree` based element, like an `array` or an `object`?

File: iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.bicep (hunk 3 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/arm/src/test/resources/checks/ResourceSpecificAdminAccountCheck/Microsoft.ContainerRegistry_registries.bicep:19]: Same question here.

**Pull request merged**